### PR TITLE
Report test duration. Fix #6482

### LINKED ISSUE #6482: Calculate & Report Durations in Composed Specs
Our duration is always 0 in the `run` method defined in
`test/shared/src/main/scala/zio/test/TestRunner.scala`

### PR DESCRIPTION
**Pull request by swoogles**

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/TestAnnotationRenderer.scala [hunk 1]
```diff
@@ -129,8 +129,8 @@ object TestAnnotationRenderer {
    * time.
    */
   val timed: TestAnnotationRenderer =
-    LeafRenderer(TestAnnotation.timing) { case (child :: ancestors) =>
+    LeafRenderer(TestAnnotation.timing) { case (child :: _) =>
       if (child.isZero) None
-      else Some(f"${child.render} (${(child.toNanos.toDouble / (child :: ancestors).last.toNanos) * 100}%2.2f%%)")
+      else Some(f"${child.render} ")
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/TestAnnotationRenderer.scala:134]: Extra space?

File: test/shared/src/main/scala/zio/test/Summary.scala [hunk 2]
```diff
@@ -16,10 +16,10 @@
 
 package zio.test
 
-import zio.ZTraceElement
+import zio.{Duration, ZTraceElement}
 import zio.stacktracer.TracingImplicits.disableAutoTrace
 
-final case class Summary(success: Int, fail: Int, ignore: Int, summary: String) {
+final case class Summary(success: Int, fail: Int, ignore: Int, summary: String, duration: Duration = Duration.Zero) {
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/Summary.scala:22]: 😱

File: test/shared/src/main/scala/zio/test/TestAnnotationRenderer.scala (hunk 1 above)
Comment (swoogles) [reply test/shared/src/main/scala/zio/test/TestAnnotationRenderer.scala]: Removed

File: test/shared/src/main/scala/zio/test/Summary.scala (hunk 2 above)
Comment (swoogles) [reply test/shared/src/main/scala/zio/test/Summary.scala:22]: image

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**